include/ginkgo/core/base/std_extensions.hpp

```diff
@@ -69,6 +70,16 @@ template <typename... Ts>
 using void_t = typename detail::make_void<Ts...>::type;
 
 
+// Disable deprecation warnings when using standard > C++14
+inline bool uncaught_exception() noexcept {
+#if __cplusplus > 201402L
+    return std::uncaught_exceptions() > 0;
+#else
+    return std::uncaught_exception();
+#endif
+}
```

Comment:
Is this strictly necessary  ? I dont think we should have this in the public header either way. Might have unexpected consequences for integrations.

Reply:
What would be the consequences? I added this because the version we use of `std::uncaught_exceptions` got deprecated in C++17, therefore rising many errors when compiling with an higher standard (which nvhpc does). Here, I select the new int version with high standards and keep the same one with lower standards. 
See: https://en.cppreference.com/w/cpp/error/uncaught_exception

Reply:
I see. Thanks for the clarification. I misunderstood this usage.